stroeer/core/v1/article.proto

```diff
@@ -9,47 +9,72 @@ option java_multiple_files = true;
 option java_package = "de.stroeer.core.v1";
 
 message Article {
-  string id = 1;
-  string web_url = 2;
-  string canonical_url = 3;
-  ContentType type = 4;
-  string section_path = 5;
-  string headline = 6;
-  map<string, string> fields = 7;
-  repeated string keywords = 8;
-  Metadata metadata = 9;
-  repeated Element elements = 10;
+  uint64 id = 1;
+  ContentType type = 2;
+  ContentSubType sub_type = 3;
+  ArticleReferences references = 4;
+  Section section = 5;
+  map<string, string> fields = 6;
+  Body body = 7;
+  Metadata metadata = 8;
+  repeated Element elements = 9;
+  repeated string keywords = 10;
   repeated Author authors = 11;
   repeated string onwards = 12;
-  Body body = 13;
+
 }
 
 message Metadata {
-  google.protobuf.Timestamp valid_from_date = 1;
-  google.protobuf.Timestamp valid_to_date = 2;
-  string state = 3;
-  google.protobuf.Timestamp transformation_date = 4;
+  string state = 1;
+  //  editorial date: Gültig von
+  google.protobuf.Timestamp valid_from = 2;
+  //  editorial date: Gültig bis
+  google.protobuf.Timestamp valid_to = 3;
+  //  editorial date: Zuletzt bearbeitet
+  google.protobuf.Timestamp last_edited = 4;
+  //  editorial date: Zuletzt publiziert
+  google.protobuf.Timestamp last_published = 5;
+  // editorial date: Erstellungsdatum
+  google.protobuf.Timestamp created = 6;
+  // editorial date: Aktualisierungsdatum
+  google.protobuf.Timestamp updated = 7;
 }
 
 message Element {
   ElementType type = 1;
-  string guid = 2;
-  repeated string relations = 3;
-  repeated Asset assets = 4;
-  string article_id = 5;
+  repeated ElementRelation relations = 2;
+  repeated Asset assets = 3;
 }
 
 message Asset {
   AssetType type = 1;
-  int32 index = 2;
-  map<string, string> fields = 4;
-  Metadata metadata = 5;
+  uint32 index = 2;
```

Comment:
why `uint32` here and article id is `uint64`?

Reply:
this is a 0-based index, where as the article id is the database id which is somehwere 100,000,000+
In Java it would be the decision between `int` and `long`